src/cmd-buildextend-metal

```diff
@@ -67,19 +72,20 @@ if [ $# -ne 0 ]; then
 fi
 
 # default to both BIOS and UEFI for non-s390x and BIOS DASD for s390x
-if [ -z "${BIOS}" ] && [ -z "${UEFI}" ] && [ -z "${DASD}" ]; then
-    BIOS=1
+if [ -z "${BIOS}" ] && [ -z "${UEFI}" ] && [ -z "${DASD}" ] && [ -z "${ZFCP}" ]; then
     if [ "${arch}" != "s390x" ]; then
+        BIOS=1
         UEFI=1
     else
         DASD=1
+        ZFCP=1
     fi
 fi
 
-if [ -n "${UEFI}" ] && [ "${arch}" = "s390x" ]; then
-    fatal "${arch} is not supported for building UEFI image"
-elif [ -n "${DASD}" ] && [ "${arch}" != "s390x" ]; then
-    fatal "${arch} is not supported for building DASD image"
+if ([ -n "${BIOS}" ] || [ -n "${UEFI}" ]) && [ "${arch}" = "s390x" ]; then
+    fatal "${arch} is not supported for building ${BIOS:+BIOS}${UEFI:+UEFI} image"
```

Comment:
Minor: how about

```
>      fatal "${arch} is not supported for building images: ${BIOS:+BIOS} ${UEFI:+UEFI}" 
```

?

(and similarly below)

Reply:
Updated. Thanks!